hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -739,48 +735,59 @@ private void internalGetSSTDiffList(
       // Traversal level/depth indicator for debug print
       int level = 1;
       while (!currentLevel.isEmpty()) {
-        LOG.debug("BFS level: {}. Current level has {} nodes.",
+        LOG.debug("Traversal level: {}. Current level has {} nodes.",
             level++, currentLevel.size());
 
+        if (level >= 1000000) {
+          LOG.error("Graph traversal level exceeded allowed maximum ({}). "
+                  + "This could be due to invalid input generating a "
+                  + "loop in the traversal path. Same SSTs found so far: {}, "
+                  + "different SSTs: {}", level, sameFiles, differentFiles);
+          // Clear output to indicate an error. Expect fall back to full diff
+          sameFiles.clear();
+          differentFiles.clear();
```

Comment:
getSSTDiffList() doesn't seem to treat this as an error.  Shouldn't it?

Reply:
Good point. done